.jenkins/release.Jenkinsfile

```diff
@@ -47,6 +50,37 @@ pipeline {
                         }
                     }
                 }
+                stage('Deploy') {
+                    agent { label "master" }
+                    options {
+                        withAWS(role: 'CI', roleAccount: '508912190628', region: 'eu-west-1')
+                    }
+                    environment {
+                        CODEARTIFACT_AUTH_TOKEN = "${sh(script: 'aws codeartifact get-authorization-token --domain adverity --domain-owner 508912190628 --query authorizationToken --output text', returnStdout: true)}".trim()
+                    }
+                    steps {
+                        checkout(
+                            scm: [
+                                $class: 'GitSCM',
+                                branches: [[name: "refs/tags/${releaseVersion}"]],
+                                doGenerateSubmoduleConfigurations: false,
+                                extensions: [
+                                        [$class: 'CleanCheckout'],
+                                        [$class: 'CloneOption', noTags: false, reference: '', shallow: false]
+                                ],
+                                userRemoteConfigs: [[url: 'git@github.com:pmedianetwork/eslint-config.git']]
+                            ]
+                        )
```

Comment:
cannot this be a part of the jobdsl?

Reply:
this is actually a nested git checkout within the job in another jenkins node

Reply:
hmm this could also be a separate job. which is triggered on tag creation

Reply:
why does this checkout itself?

Reply:
it checks out the created tag by the previous stage
I wanted to make sure the state is clean by adding a `git clean after checkout`
Also for the possibility to actually have this in a separate job in the future. When the monorepo setup will be done.. since this is rather temporary.

Reply:
but yes.. I could also do this "manually" with
```
sh 'git fetch --tags'
sh 'git checkout refs/tags/${releaseVersion}
sh 'git reset --hard'
sh 'git clean -fdx'
```

Reply:
that jenkins scm pipeline syntax makes me cringe, anything against using git directly?

Reply:
no.. it's fine..^^

Reply:
https://github.com/pmedianetwork/eslint-config/pull/87/commits/9a12deaf4baf0f1771cf3c315310a7b37d68815c